plugins/drm/src/drm-plugin.c

```diff
@@ -258,8 +258,16 @@ INTERNAL int drmp_get_monitor_info(struct surfman_plugin *plugin, surfman_monito
     }
     drmModeFreeConnector(c);
 
-    info->current_mode = &info->modes[0];
-    info->prefered_mode = &info->modes[0];
+    /* If we give back any value higher than 3000, surfman will override it.
+     * Try to find the highest resolution with both components below 3000. */
+    while (mode < modes_count - 1 &&
```

Comment:
Although it was not checked before, `modes_count == 0` is dangerous here.
This does not come from your PR, but if you would not mind adding a sanity check and returning `SURFMAN_ERROR` earlier that would be nice.